docs/service_architecture.md

```diff
@@ -166,6 +218,161 @@ keys in the HSM, but not to exfiltrate the private material.
+protected and the Notary CLI does not provide options to export them in
+plaintext.
+
+It is up to the user to choose an appropriate password, and to protect their key
+from offline brute-force attacks.
+
+The severity of the compromise of a trust collection owner/administrator's
+decrypted key depends on the type and combination of keys that were compromised
+(e.g. the snapshot key and targets key, or just the targets key).
+
+#### Possible attacks given the credentials compromised:
+
+- **Decrypted Delegation Key, only**
+
+  | Keys compromised | Malicious Content | Rollback, Freeze, Mix and Match | Denial of Service |
+  |------------------|-------------------|---------------------------------|-------------------|
+  | Delegation key   | no                | no                              | no                |
+
+- **Decrypted Delegation Key + Notary Service write-capable credentials**
+
+  | Keys compromised | Malicious Content | Rollback, Freeze, Mix and Match | Denial of Service |
+  |------------------|-------------------|---------------------------------|-------------------|
+  | Delegation key   | maybe*            | maybe*                          | maybe*            |
+
+  *If the Notary Service holds the snapshot key and the attacker has Notary Service
+  write credentials, then they have effective access to the snapshot and timestamp
+  keys because the server will generate and sign the snapshot and timestamp for them.
+
+  An attacker can add malicious content, remove legitimate content from a collection, and
+  mix up the targets in a collection, but only within the particular delegation
+  roles that the key can sign for.  Depending on the restrictions on that role,
+  they may be restricted in what type of content they can modify.  They may also
+  add or remove the capabilities of other delegation keys below it on the key hierarchy
+  (e.g. if `DelegationKey2` in the above key hierarchy were compromised, it would only be
+  able to modify the capabilities of `DelegationKey4` and `DelegationKey5`).
+
+- **Decrypted Delegation Key + Decrypted Snapshot Key, only**
+
+  | Keys compromised | Malicious Content | Rollback, Freeze, Mix and Match | Denial of Service |
+  |------------------|-------------------|---------------------------------|-------------------|
+  | Delegation key <br/> Snapshot key  | no    | no                        | no                |
+
+  The attacker does not have access to the timestamp key, which is always held by the Notary
+  Service, and will be unable to set up a malicious mirror.
+
+- **Decrypted Delegation Key + Decrypted Snapshot Key + Notary Service write-capable credentials**
+
+  | Keys compromised | Malicious Content | Rollback, Freeze, Mix and Match | Denial of Service |
+  |------------------|-------------------|---------------------------------|-------------------|
+  | Delegation key <br/> Snapshot key  | yes   | yes                       | yes               |
```

Comment:
I'd almost make the delegation key `yes` into a `limited` because delegation roles can have limited paths, what do you think?  Yes is fine too but a delegation key compromise in this combination is definitely less scary than if it was the targets key